### PR TITLE
Thellimist/pi 1194 make unicode zones work by punycoding 

### PR DESCRIPTION
**Pull request by thellimist**
I couldn't use [idn-to-ascii](http://php.net/manual/en/function.idn-to-ascii.php) function since php needs to be compiled with an extra flag. I **assumed** we don't have that access and downloaded a library which does it.

### CODE REVIEW

File: src/Cpanel/ClientActions.php [hunk 1]
```diff
@@ -60,15 +61,21 @@ public function mergeCpanelAndCFDomains()
             $cpanelDomainList = array_merge($cpanelDomainList, $getCpanelDomains['parked_domains']);
         }
 
+        $Punycode = new Punycode();
```
Comment (jwineman) [inline src/Cpanel/ClientActions.php]: `new` creates a hard coded dependency.  Can we do getter/setter DI here and write a test to ensure this method handles unicode domains correctly?

Comment (jwineman) [inline src/Cpanel/ClientActions.php]: If we ran into this issue on wordpress as well is there a possibility of moving this fix into `cloudflare-plugin-backend` so we ensure we handle unicode domains correctly across all plugins?

Comment (thellimist) [inline src/Cpanel/ClientActions.php]: The fix is dependent on `GET /zone?name` endpoint. `cloudflare-plugin-backend` only has generic calls. Max we can do is move the getter/setter there but I think it'd be better if each plugin has it separately unless we move`GET /zone?name` to `cloudflare-plugin-backend`

File: src/Cpanel/ClientActions.php [hunk 2]
```diff
@@ -32,6 +33,19 @@ public function __construct(DefaultIntegration $cpanelIntegration, APIInterface
         $this->logger = $cpanelIntegration->getLogger();
         $this->partialZoneSet = new Partial($this->cpanelAPI, $this->dataStore, $this->logger);
         $this->request = $request;
+
+        $this->setPunycoder();
+    }
+
+    public function setPunycoder($p = null) {
+        $this->punyCode = $p;
```
Comment (jwineman) [inline src/Cpanel/ClientActions.php]: Normally we put this logic in the getter but it doesn't matter, this approach works as well.

**Pull request merged**